Automattic/jetpack
# AI Featured Image: show usage information on the modal

Fixes #36743.

## Proposed changes:

* Add small component to show usage on the image generation modal
* Make the request counter be updated optimistically using the feature cost, provided by the backend data
* Update types and AI Feature action handler to properly get and save the feature costs on local store, the same way we did for the logo generator
* Disable generate buttons when the request balance is not enough for a new generation, and display a custom tooltip in this case

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor and add some content to it, so we can use it to generate an image
* Open the Jetpack sidebar, then the AI Assistant section; note the usage on the usage panel:

<img width="250" alt="Screenshot 2024-04-04 at 17 32 47" src="https://github.com/Automattic/jetpack/assets/6760046/34115677-5985-45da-92f9-18bf8365c204">

* Click the "Generate image" button to generate an image
* When the image is ready:
   * notice the usage information under the image on the modal
   * notice the information tooltip, hover the mouse over it and confirm it shows the cost of one image
   * notice that the usage is incremented optimistically on UI, by 10 units, the current cost of one image

<img width="350" alt="Screenshot 2024-04-04 at 17 53 37" src="https://github.com/Automattic/jetpack/assets/6760046/5962663d-f53c-4fa4-9e7d-0c0e9da92424">

* Experiment clicking the "Generate another image" button; the usage should be updated again:

<img width="200" alt="Screenshot 2024-04-04 at 17 54 42" src="https://github.com/Automattic/jetpack/assets/6760046/7a837368-037a-4d68-a937-2a2e2a89c69d">

* Close the modal and check the usage on the usage panel; it should be updated as well:

<img width="250" alt="Screenshot 2024-04-04 at 17 54 54" src="https://github.com/Automattic/jetpack/assets/6760046/20838e67-89e4-47a9-af97-f8b66388ad24">

---

Testing the request balance check:

* Launch a new site, connect it and keep it on the Jetpack AI free plan
* Make sure to enable beta blocks
* Go to the block editor and ask the AI Assistant for some content (example: two paragraphs about woodworking); this is important so we spend 1 request on this
* On the Jetpack sidebar, ask for one featured image by clicking on the "Generate image" button
* When the image is ready, confirm the usage counter shows 11/20, in the same way the sidebar usage counter does
* Since an image costs 10 requests and the site now has 9 available:
   * confirm that the "Generate another image" and the "Generate image" from the sidebar are disabled/greyed
   * confirm that the tooltip now says "Featured image generation costs 10 requests per image. You don't have enough requests to generate another image."
   * confirm you can still use the image you generated

<img width="600" alt="Screenshot 2024-04-05 at 17 03 05" src="https://github.com/Automattic/jetpack/assets/6760046/4ed1103f-cde3-4937-8e4a-69a1c708f404">

* confirm the "Generate image" button from the sidebar stays disabled after you close the modal, since the requests balance is still insuficient

<img width="250" alt="Screenshot 2024-04-05 at 17 04 37" src="https://github.com/Automattic/jetpack/assets/6760046/24094ae6-4036-4b2f-a12f-ca91170d564c">

* confirm things get enabled again after you upgrade to a 100 requests plan

<img width="600" alt="Screenshot 2024-04-05 at 17 06 41" src="https://github.com/Automattic/jetpack/assets/6760046/2096b6bf-9891-47da-8d24-c3de4017ac30">
